### ISSUE TITLE
Error: assignment to entry in nil map

### ISSUE THREAD

**Issue by SuperOleg39** [imgproxy/imgproxy#938]
Hello!

`imgproxy@3.7.0`, latest Docker image.

Try to use `format_quality:jpeg:70:avif:45:webp:70` preset on images, and `imgproxy` crashed with stacktrace:

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc00063fc00, {0xc00022b0b0, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc00063fc00, {0xc000038e5d, 0xe}, {0xc00022b0b0?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc00063fba0?, {0xc00026a960?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc00063fc00, {0xc0006d8080, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc00063fc00, {0xc000038e5d, 0xe}, {0xc0006d8080?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc00063fba0?, {0xc0006da690?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc00063fc00, {0xc0002b9f20, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc00063fc00, {0xc000038e5d, 0xe}, {0xc0002b9f20?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc00063fba0?, {0xc0002cb8c0?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc00063fc00, {0xc00028a240, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc00063fc00, {0xc000038489, 0xe}, {0xc00028a240?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc00063fba0?, {0xc0001d5890?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc00063fc00, {0xc0002782b0, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc00063fc00, {0xc000038e5d, 0xe}, {0xc0002782b0?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc00063fba0?, {0xc000277890?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0005bfc00, {0xc0002f1580, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0005bfc00, {0xc000038489, 0xe}, {0xc0002f1580?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0005bfba0?, {0xc0000ba870?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0006bfc00, {0xc000140160, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0006bfc00, {0xc000038e5d, 0xe}, {0xc000140160?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0006bfba0?, {0xc000605770?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0006bfc00, {0xc0003a8400, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0006bfc00, {0xc000038489, 0xe}, {0xc0003a8400?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0006bfba0?, {0xc00038b830?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0005c7c00, {0xc0001f2080, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0005c7c00, {0xc000038489, 0xe}, {0xc0001f2080?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0005c7ba0?, {0xc0001f86f0?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0006bfc00, {0xc0000a40f0, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0006bfc00, {0xc000038489, 0xe}, {0xc0000a40f0?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0006bfba0?, {0xc0002473b0?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0005bfc00, {0xc000208400, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0005bfc00, {0xc000038489, 0xe}, {0xc000208400?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0005bfba0?, {0xc000296990?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc0005c7c00, {0xc0002c2470, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc0005c7c00, {0xc000038e5d, 0xe}, {0xc0002c2470?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc0005c7ba0?, {0xc0002c5860?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/imgproxy/imgproxy/v3/options.applyFormatQualityOption(0xc00063fc00, {0xc000292be0, 0x6, 0x3?})
    /app/options/processing_options.go:574 +0xde
github.com/imgproxy/imgproxy/v3/options.applyURLOption(0xc00063fc00, {0xc000038489, 0xe}, {0xc000292be0?, 0x6, 0x6})
    /app/options/processing_options.go:920 +0xe25
github.com/imgproxy/imgproxy/v3/options.applyURLOptions(0xc00063fba0?, {0xc000205050?, 0x1, 0x0?})
    /app/options/processing_options.go:944 +0x97
github.com/imgproxy/imgproxy/v3/options.ValidatePresets()
    /app/options/presets.go:63 +0xd7
main.initialize()
    /app/main.go:59 +0x10a
main.run()
    /app/main.go:74 +0x3f
main.main()
    /app/main.go:126 +0x117
```

Maybe I made wrong preset, or some options has conflicts?

**Comment by DarthSim** [comment]
Hey @SuperOleg39!
Fixed the issue in v3.7.1

**Comment by SuperOleg39** [comment]
Thanks you a lot! 😍
